src/test/java/com/example/demo/service/book/BookServiceTest.java

```diff
@@ -0,0 +1,120 @@
+package com.example.demo.service.book;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.example.demo.dto.book.BookDto;
+import com.example.demo.dto.book.BookSearchParametersDto;
+import com.example.demo.dto.book.CreateBookRequestDto;
+import com.example.demo.mapper.BookMapper;
+import com.example.demo.model.Book;
+import com.example.demo.repository.book.BookRepository;
+import com.example.demo.repository.book.BookSpecificationBuilder;
+import com.example.demo.service.impl.BookServiceImpl;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceTest {
```

Comment:
Same comments. Try to simplify your test cases, move logic to the util class. Also, use Given-When-Then construction